[Bug]: `yarn build` sometimes fails to build types correctly

### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

all

### Expected behavior

When running `yarn build` or `yarn build:types` you would expect the build to always pass.

### Actual behavior

When running `yarn build` or `yarn build:types` sometimes you will experience an error like this:

```
packages/overlay/src/OverlayDialog.ts(16,8): error TS6305: Output file '/Users/delarre/Code/spectrum-web-components/tools/shared/src/first-focusable-in.d.ts' has not been built from source file '/Users/delarre/Code/spectrum-web-components/tools/shared/src/first-focusable-in.ts'.
```

The message is indicating that the `.d.ts` was not properly built from the current `.ts` source files. 

### Screenshots

_No response_

### What browsers are you seeing the problem in?

_No response_

### How can we reproduce this issue?

Run `yarn build:types`, change some source, and then run it again.

### Sample code that illustrates the problem

_No response_

### Logs taken while reproducing problem

_No response_

--- a/package.json
+++ b/package.json
@@ -260,7 +260,7 @@
             "clean": "if-file-deleted"
         },
         "build:types": {
-            "command": "tsc --build tsconfig-all.json",
+            "command": "tsc --build tsconfig-all.json --pretty",
             "dependencies": [
                 "process-icons",
                 "test:create",
@@ -272,7 +272,9 @@
             ],
             "output": [
                 "packages/**/*.d.ts",
+                "packages/**/tsconfig.tsbuildinfo",
                 "tools/**/*.d.ts",
+                "tools/**/tsconfig.tsbuildinfo",
                 "!**/local.d.ts",
                 "!tools/base/src/version.d.ts"
             ],

